### PR TITLE
Added github aliases to user

### PR DESCRIPTION
**Pull request by patrikfirek**

### CODE REVIEW

File: src/app/CodeRepo/lib/backends/github/index.js [hunk 1]
```diff
@@ -118,28 +119,37 @@ class GithubBackend extends AsyncEventEmitter {
...
-                    }
+    // Lookup user based on github alias
+    async _getUserByGithubLogin(githubUserName) {
+        let ref=false
+
+        if (typeof githubUserName === "string" && githubUserName.length > 0) {
+            try {
+                const client = ServiceComBus.instance.getClient("userrepo");
+                const users = await client.list("user", { "aliases.github": githubUserName});
+                if (!(users instanceof Array)) {
```
Comment (mattiasrunge) [inline src/app/CodeRepo/lib/backends/github/index.js]: This seems unnecessary, we should be able to assume that client.list always returns an array

File: src/app/CodeRepo/lib/backends/github/index.js [hunk 2]
```diff
@@ -118,28 +119,37 @@ class GithubBackend extends AsyncEventEmitter {
...
-                    for (const commit of event.payload.commits) {
-                        if (commit.author.name === userName) {
-                            return commit.author.email;
-                        }
-                    }
+    // Lookup user based on github alias
+    async _getUserByGithubLogin(githubUserName) {
+        let ref=false
+
+        if (typeof githubUserName === "string" && githubUserName.length > 0) {
```
Comment (mattiasrunge) [inline src/app/CodeRepo/lib/backends/github/index.js]: Maybe it's just style but if we do error cases first and return, we don't get so deeply nested { which in my view makes things harder to read.

File: src/app/CodeRepo/lib/backends/github/index.js [hunk 3]
```diff
@@ -118,28 +119,37 @@ class GithubBackend extends AsyncEventEmitter {
...
-            const result = await this._sendRequest(url, {}, "GET");
-            for (const event of result) {
-                if (event.type && event.type === "PushEvent" && event.payload && event.payload.commits) {
-                    for (const commit of event.payload.commits) {
-                        if (commit.author.name === userName) {
-                            return commit.author.email;
-                        }
-                    }
+    // Lookup user based on github alias
+    async _getUserByGithubLogin(githubUserName) {
```
Comment (christensson) [inline src/app/CodeRepo/lib/backends/github/index.js]: Duplicate of Revision#_getUserRef. Modify Revision#_getUserRef instead to take query as parameter and use that function.

File: src/app/UserRepo/lib/types/user.js [hunk 4]
```diff
@@ -20,6 +24,7 @@ class User extends Type {
         this.keys = [];
         this.teams = [];
         this.policyRefs = [];
+        this.aliases = [];
```
Comment (christensson) [inline src/app/UserRepo/lib/types/user.js]: Aliases is here an array. But later required to be an object...

**Pull request merged**